### PR TITLE
adding docs for creating array components

### PR DESCRIPTION
**Pull request by yoavaa**
Docs in the `array-components.md` file for how to create AutoViews components for arrays

### CODE REVIEW

File: packages/core/src/repository/components-repo.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export class ComponentsRepo {
 
     constructor(
         public name: string,
-        public getNodeType: GetNode = node => node.type
+        public getNodeType: GetNode = node => node.type as string
```
Comment (tyv) [inline packages/core/src/repository/components-repo.ts:62]: @Fer0x shouldn't this also return a symbol?

Comment (Fer0x) [reply packages/core/src/repository/components-repo.ts:62]: No, components registered with symbol not supposed to be rendered in natural way via getNodeType/getMatched, only with direct use of `getByType` method.

**Comment by denysdovhan** [comment]
I just resolved the conflict between update `master` and this branch. Nothing to worry about 👌🏻

**Review by tyv — APPROVED** [review]
just wow, thanks @yoavaa

**Review by Fer0x — APPROVED** [review]

**Pull request merged**